web/src/upload_widget.ts

```diff
@@ -135,65 +344,139 @@ export function build_direct_upload_widget(
+            });
+
+        uppy.addFile({
+            name: file.name,
+            type: file.type,
+            data: file,
+        });
+
+        const scaleButton = document.querySelector("#scale_to_fit");
+        if (scaleButton) {
+            scaleButton.addEventListener("click", () => {
+                scaled = true;
+            });
+        }
+    }
+
+    $upload_button.on("click", (e) => {
+        get_file_input().trigger("click");
+        e.preventDefault();
+    });
+
     $upload_button.on("drop", (e) => {
         const files = e.originalEvent?.dataTransfer?.files;
         if (files === null || files === undefined || files.length === 0) {
             return false;
         }
-        get_file_input()[0]!.files = files;
+        const $file_input = get_file_input();
+        if ($file_input[0]) {
+            $file_input[0].files = files;
+        }
         e.preventDefault();
         return false;
     });
 
     get_file_input().attr("accept", supported_types.toString());
     get_file_input().on("change", (e) => {
-        if (e.target.files?.[0] === undefined) {
+        if (e.target.files?.length === 0) {
             $input_error.hide();
-        } else if (e.target.files.length === 1) {
+        } else if (e.target.files?.length === 1) {
             const file = e.target.files[0];
-            if (file.size > max_file_upload_size * 1024 * 1024) {
-                $input_error.text(
-                    $t(
-                        {defaultMessage: "File size must be at most {max_file_size} MiB."},
-                        {max_file_size: max_file_upload_size},
-                    ),
-                );
-                $input_error.show();
-                clear();
-            } else if (!is_image_format(file)) {
-                $input_error.text($t({defaultMessage: "File type is not supported."}));
-                $input_error.show();
-                clear();
-            } else {
-                accept();
+            if (file) {
+                if (file.size > max_file_upload_size * 1024 * 1024) {
+                    $input_error.text(
+                        $t(
+                            {defaultMessage: "File size must be at most {max_file_size} MiB."},
+                            {max_file_size: max_file_upload_size},
+                        ),
+                    );
+                    $input_error.show();
+                    clear();
+                } else if (!is_image_format(file)) {
+                    $input_error.text($t({defaultMessage: "File type is not supported."}));
+                    $input_error.show();
+                    clear();
+                } else {
+                    accept(file);
+                }
             }
         } else {
             $input_error.text($t({defaultMessage: "Please just upload one file."}));
         }
     });
-
-    $upload_button.on("click", (e) => {
-        get_file_input().trigger("click");
-        e.preventDefault();
-    });
```

Comment:
This looks to be moved without changes; can you undo that sort of changes to keep the diff clean?

Reply:
Yes, we will try to find all of those sort of changes and undo them to make the diff clean.